src/main/java/com/example/bookwonders/mapper/OrderItemMapper.java

```diff
@@ -0,0 +1,41 @@
+package com.example.bookwonders.mapper;
+
+import com.example.bookwonders.config.MapperConfig;
+import com.example.bookwonders.dto.order.OrderItemResponseDto;
+import com.example.bookwonders.exception.EntityNotFoundException;
+import com.example.bookwonders.model.Book;
+import com.example.bookwonders.model.CartItem;
+import com.example.bookwonders.model.OrderItem;
+import com.example.bookwonders.repository.book.BookRepository;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Mapper(config = MapperConfig.class)
+public abstract class OrderItemMapper {
+    @Autowired
+    private BookRepository bookRepository;
+
+    public abstract OrderItemResponseDto toDto(OrderItem orderItem);
+
+    @Mapping(target = "id", ignore = true)
+    public abstract OrderItem cartItemToOrderItem(CartItem cartItemResponseDto);
+
+    @AfterMapping
+    public void setBookToOrderItem(@MappingTarget OrderItem orderItem,
+                                            CartItem cartItem) {
+        Book book = bookRepository.findById(cartItem.getBook().getId()).orElseThrow(() ->
+                new EntityNotFoundException("can't find book by id: "
+                        + cartItem.getBook().getId()));
+        orderItem.setPrice(book.getPrice());
+        orderItem.setBook(book);
+    }
```

Comment:
I don't really like idea of using repositories during mapping.
And especially here when we possibly create dozens of requests to DB during mapping of big orders.

Both OrderItem and CartItem has Book inside. Can we have this Book properly initialized before mapping to avoid calling of repository?